src/ui/dice.test.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import React from 'react';
+import { Dice } from './dice';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+test('basic', () => {
+  {
+    const dice = Enzyme.shallow(<Dice />);
+    expect(dice.html()).toContain('div');
+  }
+  {
+    const dice = Enzyme.shallow(<Dice className="custom" />);
+    expect(dice.html()).toContain('custom');
+  }
+  {
+    const dice = Enzyme.shallow(<Dice dotStyle={false} faceValue={1} />); // renders plain text instead of styled dots
+    expect(dice.text()).toContain(1);
+  }
+});
+
+test('state change', () => {
+  {
+    const dice = Enzyme.mount(<Dice faceValue={4} />);
+    dice.setProps({ faceValue: 1 });
+    expect(dice.state().faceValue).toBe(1);
+    dice.setProps({ faceValue: 1 });
```

Comment:
it's duplicate?

Reply:
It seems so, but that tests the if branch in the componentWillReceiveProps. Otherwise, line #36 in dice.js would remain untested since we couldn't tell if it would still change the state.

Reply:
ok